src/samples/plugins/ModularSessionKeyPlugin.sol

```diff
@@ -0,0 +1,338 @@
+                revert WrongDataLength();
+            }
+            for (uint256 i = 0; i < tempOwners.length; i++) {
+                _removeTemporaryOwner(msg.sender, tempOwners[i], allowedSelectors[i]);
+            }
+        }
+    }
+
+    /// @inheritdoc BasePlugin
+    function userOpValidationFunction(uint8 functionId, UserOperation calldata userOp, bytes32 userOpHash)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        if (functionId == uint8(FunctionId.USER_OP_VALIDATION_TEMPORARY_OWNER)) {
+            (address signer,) = (userOpHash.toEthSignedMessageHash()).tryRecover(userOp.signature);
+            bytes4 selector = bytes4(userOp.callData[0:4]);
+            bytes memory key = userOp.sender.allocateAssociatedStorageKey(0, 1);
+            StoragePointer ptr = key.associatedStorageLookup(keccak256(abi.encodePacked(signer, selector)));
+            SessionInfo storage duration = _castPtrToStruct(ptr);
+            uint48 _after = duration._after;
+            uint48 _until = duration._until;
+
+            if (_until != 0) {
+                return _packValidationData(false, _until, _after);
+            } else {
+                return _SIG_VALIDATION_FAILED;
+            }
+        }
+        revert NotImplemented();
+    }
+
+    /// @inheritdoc BasePlugin
+    function runtimeValidationFunction(uint8 functionId, address sender, uint256, bytes calldata data)
+        external
+        view
+        override
+    {
+        if (functionId == uint8(FunctionId.RUNTIME_VALIDATION_TEMPORARY_OWNER)) {
+            bytes4 selector = bytes4(data[0:4]);
+            bytes memory key = address(msg.sender).allocateAssociatedStorageKey(0, 1);
+            StoragePointer ptr = key.associatedStorageLookup(keccak256(abi.encodePacked(sender, selector)));
+            SessionInfo storage duration = _castPtrToStruct(ptr);
+            uint48 _after = duration._after;
+            uint48 _until = duration._until;
+
+            if (_until != 0) {
+                if (block.timestamp < _after || block.timestamp > _until) {
+                    revert WrongTimeRangeForSession();
+                }
+                return;
+            } else {
+                revert NotAuthorized();
+            }
+        }
+        revert NotImplemented();
+    }
+
+    /// @inheritdoc BasePlugin
+    function pluginManifest() external pure override returns (PluginManifest memory) {
+        PluginManifest memory manifest;
+
+        manifest.executionFunctions = new bytes4[](5);
+        manifest.executionFunctions[0] = this.addTemporaryOwner.selector;
+        manifest.executionFunctions[1] = this.removeTemporaryOwner.selector;
+        manifest.executionFunctions[2] = this.addTemporaryOwnerBatch.selector;
+        manifest.executionFunctions[3] = this.removeTemporaryOwnerBatch.selector;
+        manifest.executionFunctions[4] = this.getSessionDuration.selector;
```

Comment:
It may save gas to remove this view function, since the user can call the plugin directly with the account address as an argument. Pros and cons here, but we're thinking of making a recommendation against this pattern of adding view functions to the account.